tools/crm_mon.c

```diff
@@ -168,50 +178,75 @@ struct {
     int reconnect_msec;
     int fence_history_level;
     int verbose;
-    gboolean one_shot;
     gboolean daemonize;
-    gboolean group_by_node;
-    gboolean inactive_resources;
-    gboolean watch_fencing;
     gboolean show_bans;
-    gboolean print_clone_detail;
-    gboolean print_brief;
-    gboolean print_pending;
     char *pid_file;
     char *external_agent;
     char *external_recipient;
+    unsigned int mon_ops;
 } options = {
     .reconnect_msec = 5000,
-    .fence_history_level = 2
+    .fence_history_level = 2,
+    .mon_ops = mon_op_default
 };
 
 static gboolean
-reconnect_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
-    options.reconnect_msec = crm_get_msec(optarg);
+as_cgi_cb(const gchar *option_name, const gchar *optarg, gpointer data, GError **error) {
+    if (output_format != mon_output_console) {
+        g_set_error(error, G_OPTION_ERROR, CRM_EX_INVALID_PARAM, "-w requires console output format");
+        return FALSE;
+    }
+
+    output_format = mon_output_cgi;
+    options.mon_ops |= mon_op_one_shot;
```

Comment:
It'll be a pain, but rearrange the original commit's functions in the same order, for a cleaner history/diff